sim/druid/feral/feral.go

```diff
@@ -0,0 +1,101 @@
+package feral
+
+import (
+	"time"
+
+	"github.com/wowsims/tbc/sim/core"
+	"github.com/wowsims/tbc/sim/core/proto"
+	"github.com/wowsims/tbc/sim/druid"
+	"github.com/wowsims/tbc/sim/core/stats"
+)
+
+func RegisterFeralDruid() {
+	core.RegisterAgentFactory(
+		proto.Player_FeralDruid{},
+		proto.Spec_SpecFeralDruid,
+		func(character core.Character, options proto.Player) core.Agent {
+			return NewFeralDruid(character, options)
+		},
+		func(player *proto.Player, spec interface{}) {
+			playerSpec, ok := spec.(*proto.Player_FeralDruid)
+			if !ok {
+				panic("Invalid spec value for Feral Druid!")
+			}
+			player.Spec = playerSpec
+		},
+	)
+}
+
+func NewFeralDruid(character core.Character, options proto.Player) *FeralDruid {
+	feralOptions := options.GetFeralDruid()
+
+	selfBuffs := druid.SelfBuffs{}
+	if feralOptions.Options.InnervateTarget != nil {
+		selfBuffs.InnervateTarget = *feralOptions.Options.InnervateTarget
+	} else {
+		selfBuffs.InnervateTarget.TargetIndex = -1
+	}
+
+	druid := druid.New(character, selfBuffs, *feralOptions.Talents)
+	druid.CatForm = true
```

Comment:
I think druid.applyTalents() gets called within druid.New() so this assignment to CatForm is coming too late. Probably need to modify the druid.New parameters, perhaps by adding CatForm as an option to SelfBuffs (which can be renamed if you want, the name SelfBuffs isn't really appropriate anymore, DruidOptions might be better).

Reply:
Hmm, the talent stat modifications that require CatForm to be true are all being applied correctly in the Feral sim, which wouldn't be the case if applyTalents() was called first right?

Reply:
Ah whoops you're right, I forgot I refactored ApplyTalents awhile ago so that it gets called by core. This is fine then.